rclpy/test/test_parameter.py

```diff
@@ -213,6 +215,27 @@ def test_parameter_value_to_python(self):
         with pytest.raises(RuntimeError):
             parameter_value_to_python(parameter_value)
 
+    def test_parameter_dict_from_yaml_file(self):
+        yaml_string = """/param_test_target:
+            ros__parameters:
+                param_1: 1
+                param_str: string
+            """
+        expected = {
+            'param_1': Parameter('param_1', Parameter.Type.INTEGER, 1).to_parameter_msg(),
+            'param_str': Parameter('param_str', Parameter.Type.STRING, 'string').to_parameter_msg()
+        }
+
+        with NamedTemporaryFile(mode='w', delete=False) as f:
+            f.write(yaml_string)
+            f.flush()
+            f.close()
+            parameter_dict = parameter_dict_from_yaml_file(f.name)
+            os.unlink(f.name)
```

Comment:
If any of the lines inside the with-clause above raise an error, then the temporary file will not be removed. I think we can guarantee the file is destroyed by wrapping the whole thing in a try-finally:

```py
try:
    with NamedTemporaryFile(mode='w', delete=False) as f:
        ...
finally:
    os.unlink(f.name)
```

However, why not just set `delete=True` (the default)? We're doing exactly that in the `load_parameters_*` tests.